### ISSUE TITLE
Syntax highlighting for Chapel language

### ISSUE THREAD

**Issue by ben-albrecht** [jneen/rouge#687]
### Description

Add syntax support for [Chapel](chapel.cray.com) programming language

### Links / references

- [Existing highlighters for Chapel](https://github.com/chapel-lang/chapel/tree/master/highlight)

### Motivation

Chapel highlighting is currently supported on GitHub:

```chapel
proc hi() {
  writeln('Hello Rouge');
}
```

We'd like to add support to GitLab as well, which uses Rouge.

**Comment by ben-albrecht** [comment]
If there is documentation or useful examples of converting any of our existing highlighters, such as Pygments -> Rouge, I'd be happy to take a shot at it myself.

**Comment by ben-albrecht** [comment]
> If there is documentation or useful examples of converting any of our existing highlighters,

It looks like there's a guide here to follow:

https://www.rubydoc.info/github/rouge-ruby/rouge/file/docs/LexerDevelopment.md

**Comment by pyrmont** [comment]
@ben-albrecht Yeah, I added that shortly after coming on board as a maintainer. We have a PR (#1145)   with fixes that need to be added but hopefully that's helpful.

**Comment by tancnle** [comment]
Related PR https://github.com/rouge-ruby/rouge/pull/1743/files
